src/Compilers/Shared/DesktopBuildClient.cs

```diff
@@ -488,10 +495,13 @@ private static bool CheckPipeConnectionOwnership(NamedPipeClientStream pipeStrea
         /// </summary>
         protected override string GetSessionKey(BuildPaths buildPaths)
         {
-            return GetPipeNameForPath(buildPaths.ClientDirectory);
+            return GetPipeNameForPathOpt(buildPaths.ClientDirectory);
         }
 
-        internal static string GetPipeNameForPath(string compilerExeDirectory)
+        /// <returns>
+        /// Null if not enough information was found to create a valid pipe name.
+        /// </returns>
+        internal static string GetPipeNameForPathOpt(string compilerExeDirectory)
```

Comment:
I usually use `TryGet` for out var + bool return, and Opt for something that may be null. The latter is very common in Roslyn. I don't really care either way, though.